zypp/DiskUsageCounter.cc

```diff
@@ -249,11 +249,28 @@ namespace zypp
 	      continue;
 	    }
 
-	    //
-	    // check for snapshotting btrfs
-	    //
 	    if ( words[2] == "btrfs" )
 	    {
+	      //
+	      // Check whether is a btrfs nested subvolume
+	      //
+	      for ( unsigned i = 0; i < flags.size(); ++i ) {
+	        if ( flags[i].substr(0,8) == "subvolid" ) {
+	          // btrfs uses 5 (0 is a valid alias) for top-level volumes
```

Comment:
No. We're processing /proc/mounts output here; you cant rely on "subvolid=5" showing up here.
```
# cat /proc/mounts | grep btrfs
/dev/vda2 / btrfs rw,relatime,space_cache,subvolid=259,subvol=/@/.snapshots/1/snapshot 0 0
/dev/vda2 /.snapshots btrfs rw,relatime,space_cache,subvolid=258,subvol=/@/.snapshots 0 0
/dev/vda2 /var/spool btrfs rw,relatime,space_cache,subvolid=276,subvol=/@/var/spool 0 0
/dev/vda2 /var/log btrfs rw,relatime,space_cache,subvolid=274,subvol=/@/var/log 0 0
/dev/vda2 /var/opt btrfs rw,relatime,space_cache,subvolid=275,subvol=/@/var/opt 0 0
/dev/vda2 /var/tmp btrfs rw,relatime,space_cache,subvolid=277,subvol=/@/var/tmp 0 0
/dev/vda2 /var/lib/pgsql btrfs rw,relatime,space_cache,subvolid=273,subvol=/@/var/lib/pgsql 0 0
/dev/vda2 /var/lib/named btrfs rw,relatime,space_cache,subvolid=272,subvol=/@/var/lib/named 0 0
/dev/vda2 /var/lib/mysql btrfs rw,relatime,space_cache,subvolid=271,subvol=/@/var/lib/mysql 0 0
/dev/vda2 /var/lib/mariadb btrfs rw,relatime,space_cache,subvolid=270,subvol=/@/var/lib/mariadb 0 0
/dev/vda2 /usr/local btrfs rw,relatime,space_cache,subvolid=266,subvol=/@/usr/local 0 0
/dev/vda2 /var/crash btrfs rw,relatime,space_cache,subvolid=267,subvol=/@/var/crash 0 0
/dev/vda2 /tmp btrfs rw,relatime,space_cache,subvolid=265,subvol=/@/tmp 0 0
/dev/vda2 /var/lib/libvirt/images btrfs rw,relatime,space_cache,subvolid=268,subvol=/@/var/lib/libvirt/images 0 0
/dev/vda2 /srv btrfs rw,relatime,space_cache,subvolid=264,subvol=/@/srv 0 0
/dev/vda2 /opt btrfs rw,relatime,space_cache,subvolid=263,subvol=/@/opt 0 0
/dev/vda2 /boot/grub2/x86_64-efi btrfs rw,relatime,space_cache,subvolid=261,subvol=/@/boot/grub2/x86_64-efi 0 0
/dev/vda2 /boot/grub2/i386-pc btrfs rw,relatime,space_cache,subvolid=260,subvol=/@/boot/grub2/i386-pc 0 0
/dev/vda2 /home btrfs rw,relatime,space_cache,subvolid=262,subvol=/@/home 0 0
/dev/vda2 /var/lib/mailman btrfs rw,relatime,space_cache,subvolid=269,subvol=/@/var/lib/mailman 0 0
/dev/vda2 /var/spacewalk btrfs rw,relatime,space_cache,subvolid=287,subvol=/@/var/spacewalk 0 0
```